airbyte-integrations/connectors/source-netsuite-odbc/source_netsuite_odbc/streams.py

```diff
@@ -0,0 +1,279 @@
+
+from airbyte_cdk.sources.streams import Stream, IncrementalMixin
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Union, Final
+from airbyte_cdk.models import AirbyteMessage, AirbyteStream, SyncMode
+from datetime import date
+from airbyte_cdk.models import (
+    AirbyteMessage,
+    AirbyteStateMessage,
+    Type,
+    AirbyteStateType,
+    AirbyteStreamState,
+    StreamDescriptor,
+)
+
+# A stream's read method can return one of the following types:
+# Mapping[str, Any]: The content of an AirbyteRecordMessage
+# AirbyteMessage: An AirbyteMessage. Could be of any type
+StreamData = Union[Mapping[str, Any], AirbyteMessage]
+
+NETSUITE_PAGINATION_INTERVAL: Final[int] = 10000
+EARLIEST_DATE: Final[str] = date(2020, 1, 1)
+YEARS_FORWARD_LOOKING: Final[int] = 1
+STARTING_PRIMARY_KEY_VALUE: Final[int] = -1
+
+
+
+class NetsuiteODBCStream(Stream):
```

Comment:
I think this is the file that deserves a little overview/top-level comment.

Specifically, can you describe how incremental works?

My understanding is:
- `last_values_seen` is the `last_date_updated`
- `last_values_seen` acts as a pagination cursor because multiple queries will be required if there are more than 10k records in the stream_slice
- on a retry, we don't need to re-sync the full slice because the records are ordered by the primary key